maro/simulator/scenarios/vm_scheduling/zone.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from typing import List
+
+from maro.backends.frame import NodeAttribute, NodeBase, node
+
+
+@node("zones")
+class Zone(NodeBase):
+    """Zone node definition in frame."""
+    id = NodeAttribute("i2")
+    region_id = NodeAttribute("i2")
+
+    total_machine_num = NodeAttribute("i")
+    empty_machine_num = NodeAttribute("i")
+
+    def __init__(self):
+        self._id: int = 0
+        self._region_id: int = 0
+
+        self._name: str = ""
+        self._data_center_list: List[int] = []
+
+    def set_init_state(self, id: int, region_id: int):
+        """Set initialize state, that will be used after frame reset.
+
+        Args:
+            id (int): Region id.
+        """
+        self._id = id
+        self._region_id = region_id
+
+        self.reset()
+
+    def reset(self):
+        """Reset to default value."""
+        self.id = self._id
+        self.region_id = self._region_id
+
+        self.total_machine_num = 0
```

Comment:
`total_machine_num` can't be reset to 0.

since it's only assigned in `BE._init_xxx()` which will not be called again in env.reset().

You can test these like:
```
env = Env(...)

For epoch in range(2):
    while not is_done:
         # Action due to a simple rule
    # Record metrics of this epoch

    env.reset()
 
# Check if the metrics of all tested epochs are the same.

```